### DISCUSSION TITLE
Help me get into Flux2

### DISCUSSION THREAD

**Discussion by budimanjojo** [fluxcd/flux2#1532] [General]
Hello, currently I'm using plain yaml manifests for my K8s cluster (I don't use Helm nor Kustomize). Here's how my current yaml files folder structure looks like:
```
app01
  -> deployment.yaml
  -> service.yaml
  -> secret.yaml
  -> crd.yaml
app02
  -> deployment.yaml
  -> service.yaml
...
```
Now, I have managed to successfully deploy flux2 components and cloned the git repo into my system using the [Getting Started Guide](https://fluxcd.io/docs/get-started/). Here's how my current structure looks like (default everything):
```
flux
  -> flux-system
    -> kustomization.yaml
    -> gotk-components.yaml
    -> gotk-sync.yaml
```

My question is, can I just copy and paste the yaml manifests into the flux git repo and use a flux command to create the kustomization.yaml files for all the services? I read into the `flux create` cli command but I can't find any option to create kustomization file from manifest files. I want my flux repo structure to be like this (basically just copy the current structure into the default flux structure by namespace and add in `kustomization.yaml` file into each app folder):
```
flux
  -> flux-system
    -> kustomization.yaml
    -> gotk-components.yaml
    -> gotk-sync.yaml
  -> default
    -> app01
      -> deployment.yaml
      -> service.yaml
      -> secret.yaml
      -> crd.yaml
      -> kustomization.yaml
  -> kube-system
    -> app02
      -> deployment.yaml
      -> service.yaml
      -> kustomization.yaml
```

Is this possible? Thanks before :)

**Comment by stefanprodan** [comment]
There is no need to add a `kustomization.yaml` if don't plan to use Kustomize overlays and multi-cluster deployments, Flux generated one for you https://fluxcd.io/docs/components/kustomize/kustomization/#generate-kustomizationyaml

You may want to checkout how repos should be structured https://fluxcd.io/docs/guides/repository-structure/

**Reply by budimanjojo** [reply]
Thank you so much. So my general concern is not a problem at all. As for repo structure, it's too complicated for me as I don't need any staging or production separation. Is it okay? Or is if something I really need to do?

**Reply by kingdonb** [reply]
You want to add some structure to your repo as it grows, even if strict isolation is not a concern for you, because besides placing potentially excess load on the Kubernetes control plane by sync-checking more resources at once (with one giant implicit Kustomization that applies all of your YAML files at once, and they must all validate at once), you can keep things from breaking all at once. If something were to break in your staging environment for example, you don't want it to result in preventing deployments from going out in production.

Also you may find debugging validation issues more complicated when syncing all resources at once that are not related. Atomic applies are great, but it is better in general when unrelated things can't cause each other to fail; the design of Flux allows for these synchronizations to be independent and run separately, with their own separate deployment lifecycles.

The examples provided in the article are best practices, to avoid issues when it perhaps becomes time to scale your cluster or grow in unanticipated ways. The monorepo structure shown (infrastructure, apps, clusters) is important to know in Flux even if you aren't multi-tenant, have only one cluster, because of `dependsOn` ordering that you will find you may need to use if you ever have to use additional CRDs and validation together, or other reasons.

**Reply by budimanjojo** [reply]
Thanks for the detailed explanation. I have decided to go with this as I learned a better understanding of flux now, I followed this repo https://github.com/onedr0p/home-cluster for my structure:
```
flux:
  -> base
    -> kustomization.yaml
    -> gotk-components.yaml
    -> gotk-sync.yaml
  -> core
    -> service2
      -> kustomization.yaml
    -> ...
  -> crds
    -> service1
      -> kustomization.yaml
    -> ...
  -> apps
    -> default
      -> service1
        -> production
          -> kustomization.yaml
        -> testing
          -> kustomization.yaml
    -> kube-system
      -> ...
```
Hope this is a good approach :)
